### DISCUSSION TITLE
Enforce elementary stylesheets globally (again)?

### DISCUSSION THREAD

**Discussion by dominichayesferen** [elementary/granite#638] [General]
For context reasons, this is largely in reference to https://github.com/elementary/flatpak-platform/issues/72, and is more-so just a gentle prod towards coming to a solution to that issue.

Right now, the solutions that I know of that have been explored are:

- https://github.com/elementary/flatpak-platform/pull/30, which has stalled
- https://github.com/elementary/granite/pull/486, but it was reverted in https://github.com/elementary/granite/pull/501

Given the existence of LibAdwaita and such already doing it, and how broken Granite applications can end up without the elementary stylesheet, as proven by https://github.com/elementary/flatpak-platform/issues/72, I think it's time to re-explore #486 as a default for all Granite applications, even if Flatpak or not.

Personally, I would like to make AppCenter accessible in Feren OS, soon, but the issue illustrated in https://github.com/elementary/flatpak-platform/issues/72 currently prevents me from doing so*, for most of AppCenter (some self-enforce elementary styling, which makes them work just fine), so I thought I would just consolidate all of this into a Discussion to potentially get the ball rolling on this issue.

*Without a fix on Granite's side, it's either the applications be visually broken in most cases, or the applications be forced to use io.elementary.stylesheet.blueberry but still have missing icons and suffer a complete inability to use dark mode - it's a lose-lose situation unless addressed by applications/Granite themselves.

**Comment by danirabbit** [comment]
Yeah I think ultimately the style sheet and icons should be merged with Granite the same as in Adwaita. It would make things a lot more straightforward for developers and ensure cross platform portability

I've also wondered if we should bring back icon name constants like we used to have with "Gtk.Stock", but that's probably a different discussion.

**Reply by dominichayesferen** [reply]
> I've also wondered if we should bring back icon name constants like we used to have with "Gtk.Stock", but that's probably a different discussion.

I mean, that could always be considered after the fact.


I suppose, either way... the ultimate question is: Who's gonna do the merge request to merge stylesheet and icons into Granite, and when, at this rate (I mean, given the likelihood of there already being an agreement about it needing to be done)?

**Reply by danirabbit** [reply]
I kinda started a draft the other day, but it's not the most high priority task I have right now https://github.com/elementary/granite/pull/631

Developers can always hard code styles themselves in the meantime as in: https://github.com/elementary/music/blob/00ffe09875b6f5b7a04859238bec8b05f2921f6c/src/Application.vala#L77

**Comment by dominichayesferen** [comment]
https://github.com/elementary/granite/pull/639
